src/server/game/Spells/Spell.cpp

```diff
@@ -5186,7 +5186,7 @@ SpellCastResult Spell::CheckCast(bool strict, uint32* param1 /*= nullptr*/, uint
     {
         // Check explicit target for m_originalCaster - todo: get rid of such workarounds
         WorldObject* caster = m_caster;
-        if (m_originalCaster)
+        if (m_originalCaster && !caster->ToGameObject())
```

Comment:
could you please add a code comment here to explain why the check is needed and what it does ?